eks.tf

```
@@ -113,13 +113,13 @@ module "main" {
     for key, g in var.self_managed_node_groups :
     key => {
-      platform              = coalesce(g.platform, "linux")                                                        // Platform is optional, linux is used if omitted, also can be bottlerocket or windows https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/user_data.md#user-data--bootstrapping 
-      ami_id                = data.aws_ami.workers[key].image_id                                                   // The ID of the AMI to use for worker nodes.
-      create_security_group = false                                                                                // Don't create a dedicated security group. A common one is used instead.
-      desired_size          = g.min_nodes                                                                          // Set the desired size of the worker group to the minimum.
-      key_name              = aws_key_pair.ssh_access.key_name                                                     // The name of the SSH key to use for the nodes.
-      bootstrap_extra_args  = g.platform == "bottlerocket" ? "" : "--kubelet-extra-args '${g.kubelet_extra_args}'" // The set of extra arguments to the bootstrap script. Used to pass extra flags to the kubelet, and namely to set labels and taints. Ignored for bottlerocket.
-      iam_role_additional_policies = {                                                                             // The set of additional policies to add to the worker group IAM role.
+      platform              = coalesce(g.platform, "linux")                                                                          // Platform is optional, linux is used if omitted, also can be bottlerocket or windows https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/user_data.md#user-data--bootstrapping 
+      ami_id                = data.aws_ami.workers[key].image_id                                                                     // The ID of the AMI to use for worker nodes.
+      create_security_group = false                                                                                                  // Don't create a dedicated security group. A common one is used instead.
+      desired_size          = g.min_nodes                                                                                            // Set the desired size of the worker group to the minimum.
+      key_name              = aws_key_pair.ssh_access.key_name                                                                       // The name of the SSH key to use for the nodes.
+      bootstrap_extra_args  = g.platform == "bottlerocket" ? g.kubelet_extra_args : "--kubelet-extra-args '${g.kubelet_extra_args}'" // The set of extra arguments to the bootstrap script. Used to pass extra flags to the kubelet, and namely to set labels and taints. Ignored for bottlerocket.
```

Are you sure that's how it works?
Please try it out with something like `--max-pods=23` and then check if that worked or not.
In my opinion it requires `--kubelet-extra-args` for bottlerocket as well.

So I'd suggest:
```suggestion
      bootstrap_extra_args  = "--kubelet-extra-args '${g.kubelet_extra_args}'" // The set of extra arguments to the bootstrap script. Used to pass extra flags to the kubelet, and namely to set labels and taints. Ignored for bottlerocket.
```

---

https://bottlerocket.dev/en/os/1.19.x/api/settings/kubernetes/#max-pods

that's what it says from bottlerocket doc, and you can see it's nothing to do with passing the args to kubelet config

if you look at here you will see that's how you configure the kube setting for bottlerocket from upstream eks module

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v20.3.0/examples/self_managed_node_group/main.tf#L79-L102

A normal linux ec2 AMI uses the cloud init with kubelet to join the cluster, but the bottlerocket is not the same.

---

Interesting that they switched to TOML for bottlerocket.
In that case I recommend putting a note into the comment.

```suggestion
      bootstrap_extra_args  = g.platform == "bottlerocket" ? g.kubelet_extra_args : "--kubelet-extra-args '${g.kubelet_extra_args}'" // The set of extra arguments to the bootstrap script. Used to pass extra flags to the kubelet, and namely to set labels and taints. For bottlerocket this needs to be a TOML.
```